### PR TITLE
Add support for cpu.processors HW requirement to virtual plugin

### PR DESCRIPTION
**Pull request by happz**
Pull Request Checklist

Implements https://github.com/teemtee/tmt/issues/2941

* [x] implement the feature
* [ ] write the documentation
* [ ] extend the test coverage
* [ ] mention the version
* [ ] include a release note

### CODE REVIEW

File: tmt/steps/provision/testcloud.py [hunk 1]
```diff
@@ -531,6 +541,79 @@ def _generate_disk_filepaths() -> Iterator[Path]:
             )
 
 
+def _apply_cpu_processors(
+        hardware: Optional[tmt.hardware.Hardware],
+        domain: 'DomainConfiguration',
+        logger: tmt.log.Logger) -> None:
+    """ Apply ``tpm`` constraint to given VM domain """
```
Comment (qcheng-redhat) [inline tmt/steps/provision/testcloud.py:548]: Should be cpu.processors instead of tpm here ?

**Comment by frantisekz** [comment]
LGTM from testcloud perspective (apart from the mentioned issue by @qcheng-redhat ).

File: tmt/steps/provision/testcloud.py (hunk 1 above)
Comment (happz) [reply tmt/steps/provision/testcloud.py]: Correct, fixed in f72c0081

**Review by frantisekz — APPROVED** [review]

File: tmt/steps/provision/testcloud.py [hunk 2]
```diff
@@ -531,6 +541,79 @@ def _generate_disk_filepaths() -> Iterator[Path]:
...
 
+def _apply_hw_cpu_processors(
+        hardware: Optional[tmt.hardware.Hardware],
+        domain: 'DomainConfiguration',
+        logger: tmt.log.Logger) -> None:
+    """ Apply ``cpu.processors`` constraint to given VM domain """
+
+    domain.cpu_count = DEFAULT_CPU_COUNT
+
+    if not hardware or not hardware.constraint:
```
Comment (thrix) [inline tmt/steps/provision/testcloud.py:552]: @happz I am wondering, should this general check be somewhere before this function is even called?

**Review by thrix — APPROVED** [review]
looks clean otherwise

**Comment by happz** [comment]
Unrelated failures, merging

**Pull request merged**